### ISSUE TITLE
Use SPDX license identifiers

### ISSUE THREAD

**Issue by jaakristioja** [bibletime/bibletime#249]
We should add SPDX license identifiers to our project. At least to all source files with copyright headers.

```c++
// SPDX-License-Identifier: GPL-2.0-only
```

For more information: https://spdx.dev/ids/

PS: Are we GPL-2.0-only or GPL-2.0-or-later? O_o

**Comment by greg-hellings** [comment]
It would make sense for us to be GPL-2.0-only, because that's what SWORD is.

**Comment by jaakristioja** [comment]
What would make sense or what we want BibleTime to be are different things from what it actually is. https://github.com/bibletime/historic-bibletime-cvs/commit/a3edb9d25a67a79b9d768eb5ed21be29bf79b585 seems to indicate that we are GPL-2.0-or-later. The licenses of libraries we link against might change in the future, hence it is safer not to restrict ourselves to GPL version 2.0 only. And if BibleTime is licensed under GPL-2.0 or later, just flipping to 2.0-only would likely require explicit consent from most of our developers (both present and past).

**Comment by greg-hellings** [comment]
The way you worded your question about which were are made me think you had found evidence we are 2.0-only. That's why I said it would make sense if we are, because we came from the Sword ecosystem.

Looking at the first random source code file, I see

```c
* Copyright 1999-2020 by the BibleTime developers.
* The BibleTime source code is licensed under the GNU General Public License
* version 2.0.
```

Same goes for FILE_HEADER.cpp.in

So I would guess that we're 2.0 and not 2.0+

**Comment by jaakristioja** [comment]
Yes, but I'm fairly certain that regardless of SWORD, the BibleTime code was GPL-2.0-or-later before commit https://github.com/bibletime/historic-bibletime-cvs/commit/a3edb9d25a67a79b9d768eb5ed21be29bf79b585. I don't think one can just easily arbitrarily switch to GPL-2.0-only. As far as I know, regardless of whether BibleTime is GPL-2.0-only or GPL-2.0-or-later, as long as we link with SWORD and if SWORD is GPL-2.0-only, then we can not use GPL-3.0 the end product as a whole is GPL-2.0-only. However, I'm not an expert on these matters.

**Comment by jaakristioja** [comment]
As far as code I have myself written for BibleTime is concerned, I'm willing to relicense it under a more permissive license (e.g. MIT, BSD or similar) should the need arise (just ask).
